private/bufpkg/bufmodule/bufmodulebuild/module_file_set_builder.go

```diff
@@ -59,12 +57,8 @@ func (m *moduleFileSetBuilder) build(
 	workspace bufmodule.Workspace,
 ) (bufmodule.ModuleFileSet, error) {
 	var dependencyModules []bufmodule.Module
-	hashes := make(map[string]struct{})
-	moduleHash, err := protoPathsHash(ctx, module)
-	if err != nil {
-		return nil, err
-	}
-	hashes[moduleHash] = struct{}{}
+	moduleIds := make(map[string]struct{})
+	moduleIds[module.ID()] = struct{}{}
```

Comment:
What if `ID` is empty?

Reply:
Added `getModuleIdentifier` to use path hash as a fallback